backend/src/pdf_gen/models/model_na_31_2.rs

```diff
@@ -41,6 +45,66 @@ impl ModelNa31_2Summary {
             political_group_votes: vec![],
         }
     }
+
+    pub fn from_results(
+        election: &Election,
+        results: &[(PollingStation, PollingStationResults)],
+    ) -> Result<ModelNa31_2Summary, APIError> {
+        // running totals
+        let mut totals = ModelNa31_2Summary::zero();
+
+        // initialize political group votes to zero
+        if let Some(pgs) = &election.political_groups {
+            for group in pgs.iter() {
+                totals.political_group_votes.push(PoliticalGroupVotes {
+                    number: group.number,
+                    total: 0,
+                    candidate_votes: group
+                        .candidates
+                        .iter()
+                        .map(|c| polling_station::CandidateVotes {
+                            number: c.number,
+                            votes: 0,
+                        })
+                        .collect(),
+                });
+            }
+        }
+
+        // loop over results and add them to the running total
+        for (polling_station, result) in results {
+            // add voters and votes to the total
+            totals.voters_counts += &result.voters_counts;
```

Comment:
If recounted, we need to use the recounted voters counts instead.

Reply:
Adding up results for the entire election is the core functionality of our software. 

Feels a bit odd to do the addition inside `/pfd_gen` and inside a specific model. Shouldn't this be in a dedicated module (that is also reused for generating eml's?

Reply:
Agreed! I don't think its necessary to move the code at this point already, but we should keep it in mind.

Reply:
Maybe put in into a ticket, so we don't forget?

Reply:
I'd hope/expect that as soon as we do a calculation that's not for an Na31-2, moving that code is the obvious choice, without needing a ticket to remember.

Reply:
🤞

Reply:
Moving it to a more generic place comes naturally as more functionality is added I would say. As a part of KISS (and for the little time we currently have) I would try and prevent doing more than is necessary for the functionality we're currently adding. I'd imagine that the EML output and adding other models will have other consequences that we currently can't foresee anyway.

Reply:
I see what you mean @rnijveld , but the section for recounts and differences is unfortunately the same, see:
![Screenshot from 2024-09-12 12-27-09](https://github.com/user-attachments/assets/1971b089-5964-42f2-8506-b5d821e756c8)
![Screenshot from 2024-09-12 12-27-26](https://github.com/user-attachments/assets/89b5eb88-f3fc-4ff3-bf7b-07b4099e1620)

We could add some more information in the comments though, what do you think @praseodym @jschuurk-kr ?

Reply:
Improved in 3ddc7e0de5c21401cb8c36991b427a1609c37e93, let me know what you think!

Reply:
Looks good to me! <sub>I can't give approval on the PR though because I'm the author of it</sub>